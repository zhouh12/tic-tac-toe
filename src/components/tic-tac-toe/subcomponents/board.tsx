import React, { ReactNode } from 'react';
import { observer } from 'mobx-react';
import { SquareValue } from '../../../types/square-value-config';
import { Square } from './square';

export interface BoardProps {
  squares: SquareValue[];
  onClick: (index: number) => void;
}

// TODO: in real project, we can create a help function such as bemClassBuilder
// in order to name block or element, right now just hard-coded
const b = 'board__row';

/**
 * A component renders 9 squares.
 */
export const Board: React.FC<BoardProps> = observer((props) => {
  const renderSquare = (i: number): ReactNode => {
    return <Square value={props.squares[i]} onClick={() => props.onClick(i)} />;
  };

  return (
    <div>
      <div className={b}>
        {renderSquare(0)}
        {renderSquare(1)}
        {renderSquare(2)}
      </div>
      <div className={b}>
        {renderSquare(3)}
        {renderSquare(4)}
        {renderSquare(5)}
      </div>
      <div className={b}>
        {renderSquare(6)}
        {renderSquare(7)}
        {renderSquare(8)}
      </div>
    </div>
  );
});
