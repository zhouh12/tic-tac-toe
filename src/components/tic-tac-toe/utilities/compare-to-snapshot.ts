import { ReactElement } from 'react';
import { create, TestRendererOptions } from 'react-test-renderer';

/**
 * A helper function creates snapshot test
 */
export function compareToSnapshot(
  nextElement: ReactElement<any>,
  options?: TestRendererOptions
) {
  const tree = create(nextElement, options);
  expect(tree).toMatchSnapshot();
  tree.unmount();
}
