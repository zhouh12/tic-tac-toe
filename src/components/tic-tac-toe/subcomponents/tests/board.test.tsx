import { compareToSnapshot } from '../../utilities/compare-to-snapshot';
import { Board, BoardProps } from '../board';
import React from 'react';

describe('Board', () => {
  const onClickFunc = jest.fn();
  const testProps: BoardProps = {
    squares: ['X'],
    onClick: onClickFunc,
  };

  it('renders correctly', () => {
    compareToSnapshot(<Board {...testProps} />);
  });
});
