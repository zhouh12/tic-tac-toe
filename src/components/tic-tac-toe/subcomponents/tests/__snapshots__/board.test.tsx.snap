// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders correctly 1`] = `
<div>
  <div
    className="board__row"
  >
    <button
      className="square"
      onClick={[Function]}
    >
      X
    </button>
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board__row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
  <div
    className="board__row"
  >
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
    <button
      className="square"
      onClick={[Function]}
    />
  </div>
</div>
`;
